cms/sitemaps/cms_sitemap.py

```diff
@@ -45,13 +45,17 @@ def items(self):
         return all_pages
 
     def lastmod(self, page):
-        from cms.models import Placeholder, CMSPlugin
-        placeholders_qs = Placeholder.objects.filter(page=page)
-        placeholders_ids = list(placeholders_qs.values_list('id', flat=True))
-        plugins_qs = CMSPlugin.objects.filter(placeholder__in=placeholders_ids)
-        latest_plg_mod_date = plugins_qs.order_by(
-            '-changed_date').values_list('changed_date', flat=True)[:1]
         mod_dates = [page.changed_date, page.publication_date]
+        latest_plg_mod_date = None
+        with connection.cursor() as cursor:
+            cursor.execute(
+                "select cmsplg.changed_date from cms_cmsplugin cmsplg " +
+                "inner join cms_page_placeholders cmsp " +
+                "on cmsp.id=cmsplg.placeholder_id " +
+                "where cmsp.page_id = %s " +
+                "order by cmsplg.changed_date desc limit 1", [page.id])
+            latest_plg_mod_date = cursor.fetchone()
+
```

Comment:
`CMSPlugin.objects.filter(placeholder__page=page).only('changed_date').order_by('changed_date')[:1]` produces very similar SQL:

```SQL
SELECT `cms_cmsplugin`.`id`, `cms_cmsplugin`.`changed_date` 
FROM `cms_cmsplugin` 
INNER JOIN `cms_placeholder` ON ( `cms_cmsplugin`.`placeholder_id` = `cms_placeholder`.`id` ) 
INNER JOIN `cms_page_placeholders` ON ( `cms_placeholder`.`id` = `cms_page_placeholders`.`placeholder_id` ) 
WHERE `cms_page_placeholders`.`page_id` = ? 
ORDER BY `cms_cmsplugin`.`changed_date` ASC LIMIT 1
```

The difference is that there's a useless inner join of `cms_placeholder` instead of jumping straight to `cams_page_placeholder`. What's the performance difference? I don't think we should jump to writing SQL unless the performance is dramatic. Even then, I think the SQL needs to be in models.py and cms_sitemap.py should only call that method from the custom model manager.